npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb

```diff
@@ -169,6 +170,18 @@ def guess_yarn_version
         Helpers.yarn_version_numeric(yarn_lock)
       end
 
+      def pnpm_version
+        return @pnpm_version if defined?(@pnpm_version)
+
+        @pnpm_version = package_manager.locked_version("pnpm") || guess_pnpm_version
+      end
+
+      def guess_pnpm_version
```

Comment:
I really like the refactor/rename you did with this method, it's so much more clear the intent. 👍